crmsh/bootstrap.py

```diff
@@ -989,9 +987,15 @@ def append_to_remote_file(fromfile, remote_node, tofile):
     """
     Append content of fromfile to tofile on remote_node
     """
+    err_details_string = """
+    crmsh has no way to help you to setup up passwordless ssh among nodes at this time. 
+    As the hint, likely, `PasswordAuthentication` is 'no' in /etc/ssh/sshd_config. 
+    Given in this case, users must setup passwordless ssh beforehand, or change it to 'yes' and manage passwords properly
+    """
     cmd = "cat {} | ssh -oStrictHostKeyChecking=no root@{} 'cat >> {}'".format(fromfile, remote_node, tofile)
-    if not invoke(cmd):
-        error("Failed to run \"{}\"".format(cmd))
+    rc, _, err = invoke(cmd)
+    if not rc:
+        error("Failed to append contents of {} to {}: {}\n{}".format(fromfile, remote_node, err, err_details_string))
```

Comment:
This improvement is much more informative to the end user. One minor comment, how about to tweak the code a little:

error("Failed to append contents of {} to {}\n    \"{}\"\n{}".format(fromfile, remote_node, err, err_details_string))

hence, to put the raw error information in another dedicated line, ie,

ERROR: cluster.join: Failed to append contents of /root/.ssh/id_rsa.pub to crmsh15sp2-1
   "root@crmsh15sp2-1: Permission denied (publickey)."

Reply:
OK, changed